.github/workflows/ci.yml

```diff
@@ -0,0 +1,187 @@
+---
+name: "CI"
+on:  # yamllint disable
+  - "push"
+  - "pull_request"
+
+jobs:
+  black:
+    runs-on: "ubuntu-20.04"
+    env:
+      INVOKE_NAUTOBOT_LOCAL: "True"
+    steps:
+      - name: "Check out repository code"
+        uses: "actions/checkout@v2"
+      - name: "Setup environment"
+        uses: "networktocode/gh-action-setup-poetry-environment@v2"
+      - name: "Linting: black"
+        run: "poetry run invoke black"
+  flake8:
+    runs-on: "ubuntu-20.04"
+    env:
+      INVOKE_NAUTOBOT_LOCAL: "True"
+    steps:
+      - name: "Check out repository code"
+        uses: "actions/checkout@v2"
+      - name: "Setup environment"
+        uses: "networktocode/gh-action-setup-poetry-environment@v2"
+      - name: "Linting: flake8"
+        run: "poetry run invoke flake8"
+  hadolint:
+    runs-on: "ubuntu-20.04"
+    env:
+      INVOKE_NAUTOBOT_LOCAL: "True"
+    steps:
+      - name: "Check out repository code"
+        uses: "actions/checkout@v2"
+      - name: "Run Hadolint"
+        uses: "hadolint/hadolint-action@v1.6.0"
+        with:
+          dockerfile: "docker/Dockerfile"
+  check-migrations:
+    runs-on: "ubuntu-20.04"
+    env:
+      INVOKE_NAUTOBOT_LOCAL: "True"
+      NAUTOBOT_DB_USER: "nautobot"
+      NAUTOBOT_DB_PASSWORD: "decinablesprewad"
+      NAUTOBOT_DB_ENGINE: "django.db.backends.postgresql"
+    services:
+      postgres:
+        image: "postgres:13"
+        env:
+          POSTGRES_USER: "nautobot"
+          POSTGRES_PASSWORD: "decinablesprewad"
+          POSTGRES_DB: "nautobot"
+        ports:
+          - "5432:5432"
+    steps:
+      - name: "Check out repository code"
+        uses: "actions/checkout@v2"
+      - name: "Setup environment"
+        uses: "networktocode/gh-action-setup-poetry-environment@v2"
+      - name: "Run check migrations"
+        run: "poetry run invoke check-migrations"
+    needs:
+      - "black"
+      - "flake8"
+      - "hadolint"
+  tests-postgres:
+    runs-on: "ubuntu-20.04"
+    strategy:
+      fail-fast: true
+      matrix:
+        python-version: [ "3.6", "3.7", "3.8", "3.9" ]
+    env:
+      INVOKE_NAUTOBOT_LOCAL: "True"
+      NAUTOBOT_DB_USER: "nautobot"
+      NAUTOBOT_DB_PASSWORD: "decinablesprewad"
+      NAUTOBOT_DB_ENGINE: "django.db.backends.postgresql"
+    services:
+      postgres:
+        image: "postgres:13-alpine"
```

Comment:
Should we be using `postgres:13-alpine` for the `check-migrations` and `integration-tests` actions as well? 

I'm not clear as to why @nniehoff suggested making this change in the first place though - especially given that our dev environment uses `postgres:13` presently?

Reply:
yes they all should use `-alpine` if possible.  The smaller image will pull down significantly faster and reduce the overall testing time.  It probably wouldn't hurt to switch the dev env to -alpine but not sure that's in the scope of this PR

Reply:
Correct. I only updated on the `tests-postgres`. Fixed.